### PR TITLE
docs: ko: update `_toctree.yml`

### PR DESCRIPTION
**Pull request by HanNayeoniee**
# What does this PR do?

Part of https://github.com/huggingface/transformers/issues/20179
Initial version is in https://github.com/huggingface/transformers/pull/22581
Updated `_toctree.yml` according to https://github.com/huggingface/transformers/pull/23049

This PR restructures TOC for the documentation.
Here's the scope of the restructure:
a) TOC is sorted from “beginner” topics to more advanced
b) Some topics have been renamed
c) Task Guides are collapsed by default and now are on on the same level (currently NLP task guides are hidden, and not aligned with other modalities)
d) “General usage” has been renamed to “Developer Guides”
e) Benchmarks, notebooks, and community resources have been moved under Developer Guides
f) “Converting from TensorFlow checkpoints” and "Migrating from previous packages" pages removed

To dos:
Some topics have been renamed to be concise and more descriptive. Needs to translate to Korean.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Initial)

Team PseudoLab, may you please review this PR? @0525hhgus, @KIHOON71, @sim-so, @gabrielwithappy, @HanNayeoniee, @wonhyeongseo, @jungnerd

## Who can review? (Final)

@sgugger, @ArthurZucker, @eunseojo May you please review this PR?

### CODE REVIEW

File: docs/source/ko/_toctree.yml [hunk 1]
```diff
@@ -63,37 +40,66 @@
...
+      title: ONNX로 내보내기
     - local: in_translation
-      title: (번역중) Object detection
+      title: (번역중) Export to TorchScript
     - local: in_translation
-      title: (번역중) Zero-shot object detection
-    - local: tasks/zero_shot_image_classification
-      title: 제로샷(zero-shot) 이미지 분류
+      title: (번역중) Benchmarks
+    - local: notebooks
```
Comment (0525hhgus) [inline docs/source/ko/_toctree.yml:96]: 여기 부분이 `in_translation` 대신 `notebooks`로 되어 있습니다!

Comment (HanNayeoniee) [reply docs/source/ko/_toctree.yml:96]: 어머... 복붙하면서 그대로 가져왔나봐요. 감사합니다!!

Comment (wonhyeongseo) [reply docs/source/ko/_toctree.yml:96]: 제가 작업한 걸 참고/복붙하셔도 좋아요. #23113 
복붙 확인 후에 `closing in favor of `로 중복 PR을 닫겠습니다.

Comment (HanNayeoniee) [reply docs/source/ko/_toctree.yml]: @wonhyeongseo 
넵 감사합니다! 저희 중복된 작업을 하고 있었군요..

**Comment by wonhyeongseo** [comment]
`ko/notebook.mdx`가 https://github.com/huggingface/transformers/pull/22670 에 추가됐기 때문에 오류가 납니다. 삭제해주시면 안 날거예요.

**Review by sgugger — APPROVED** [review]
@MKhalusova Could you double-check this matches your latest changes?
Thanks.

**Comment by MKhalusova** [comment]
> @MKhalusova Could you double-check this matches your latest changes?
> Thanks.

From what I can see it does match the changes, the structure is the same as in my changes. I can't verify the translation of the titles that were renamed, but it seems to be matching too.

**Review by MKhalusova — APPROVED** [review]
Thank you for implementing the TOC restructure in the translated version of the docs!

**Pull request merged**